File: php/features.php
<?php
declare(strict_types=1);

// RealtimeWeather SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RealtimeWeatherFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RealtimeWeatherBaseFeature();
            case "test":
                return new RealtimeWeatherTestFeature();
            default:
                return new RealtimeWeatherBaseFeature();
        }
    }
}

File: py/realtimeweather_sdk/features.py
# RealtimeWeather SDK feature factory

from realtimeweather_sdk.feature.base_feature import RealtimeWeatherBaseFeature
from realtimeweather_sdk.feature.test_feature import RealtimeWeatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RealtimeWeatherBaseFeature(),
        "test": lambda: RealtimeWeatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
